lib/python/qmk/cli/multibuild.py

```diff
@@ -80,7 +80,7 @@ def multibuild(cli):
                 )
                 # yapf: enable
 
-    cli.run([make_cmd, '-j', str(cli.args.parallel), '-f', makefile.as_posix(), 'all'], capture_output=False, stdin=DEVNULL)
+    cli.run([make_cmd, *get_make_parallel_args(cli.args.parallel), '-f', makefile.as_posix(), 'all'], capture_output=False, stdin=DEVNULL)
```

Comment:
We can just add `--output-sync` to everything. We should also use the long arg so people don't have to look it up when they're trying to figure out the command.

```suggestion
    cli.run([make_cmd, *get_make_parallel_args(cli.args.parallel), '--output-sync', '-f', makefile.as_posix(), 'all'], capture_output=False, stdin=DEVNULL)
```

Reply:
I originally wanted to skip the `--output-sync` parameter in the default case (`parallel=1`). I originally thought, that `--output-sync` would still buffer the `stderr`/`stdout` even when `jobs=1`. Now I am not 100% sure, but I still don't see much sense in adding this option, if it doesn't do anything in this case. When `--jobs=1`, the output synchronization isn't required, and we probably don't want to clutter the args.

Also, IMO, even if we decide to add it unconditionally, this option should still be kept as a part of `get_make_parallel_args` output so that we DRY and so that we don't accidentally drop this option during refactoring or when adding new functionality.

I am fine with changing it to the long argument version. Might also want to change `-j` to `--jobs` while we're at it.

Reply:
We aren't likely to remove that option in the future, too many collabs use parallel building. The lack of output sync has lowkey been bugging us, but not enough to file a PR for. I'm not terribly worried about arg cluttering, we have places with much longer arglists that we get along ok with.

Good point on `--jobs`, more consistency is always desirable.

Reply:
Added the long options. Kept the location, where `--output-sync` is defined, as well as skipping it with `--jobs=1` for now. I still don't see the point of keeping a useless option, but I'll add it, if you insist.

Reply:
@skullydazed Considering, that there is some potential breakage with `--output-sync` as seen with the `flash` problems, I am even more convinced, that we should skip the `--output-sync` option when `--jobs=1`. That way, even if we've missed some edge-case where `--output-sync` causes problems, the user can just disable parallel processing and get 100% the old behaviour.

Also, I would like to close this change request, if that's alright with you (unless I've missed something, or you would like to request further changes).